docs/architecture/cloud-native/relational-vs-nosql-data.md

```diff
@@ -44,21 +44,23 @@ The theorem states that distributed data systems will offer a trade-off between
+NoSQL databases typically support high availability and partition tolerance. They scale out horizontally, often across commodity servers. This approach provides tremendous availability, both within and across geographical regions at a reduced cost. You partition and replicate data across these machines, or nodes, providing redundancy and fault tolerance. Consistency is typically tuned through consensus protocols or quorum mechanisms - which provides finer granularity of control when navigating tradeoffs between the hard extremes typically presented with tuning synchronous vs asynchronous replication in relational systems.
 
-If data replicas were to lose connectivity in a "highly available" NoSQL database cluster, you could still complete a write operation to the database. The database cluster would allow the write operation and update each data replica as it becomes available.
+If data replicas were to lose connectivity in a "highly available" NoSQL database cluster, you could still complete a write operation to the database. The database cluster would allow the write operation and update each data replica as it becomes available. NoSQL database that support multiple writable replicas can further strengthen high availability by avoiding the need for failover when optimizing recovery time objective.
 
-This kind of result is known as eventual consistency, a characteristic of distributed data systems where ACID transactions aren't supported. It's a brief delay between the update of a data item and time that it takes to propagate that update to each of the replica nodes. Under normal conditions, the lag is typically short, but can increase when problems arise. For example, what would happen if you were to update a product item in a NoSQL database in the United States and query that same data item from a replica node in Europe? You would receive the earlier product information, until the cluster updates the European node with the product change. By immediately returning a query result and not waiting for all replica nodes to update, you gain enormous scale and volume, but with the possibility of presenting older data.
+NoSQL databases typically implement partitioning (or sharding) as a native tenant to its system design. Partition management is often built-in to the database, and routing is achieved through placement hints - often called partition keys or shard keys. Flexible data models enable the NoSQL databases to lower the burden of schema management and improve availablity when deploying application updates that require data model changes.
```

Comment:
The original text here doesn't seem to understand what consistency means...

It confuses consistency in the context of ACID which has nothing to do with eventual consistency.

Eventual consistency speaks on the tradeoffs of consistency w.r.t. replication across two servers - and has nothing to do with transaction scope.

Consistency w.r.t. ACID speaks on the trade offs with respect to concurrent requests happening on the same dataset - and has nothing to do with replication (it applies to single node setups equally to replicated setups).

The document claims NoSQL systems don't support ACID transactions - which is factually incorrect. We can prove this be example using popular NoSQL database systems. How to perform ACID transactions in various NoSQL database systems:
Cosmos DB: https://devblogs.microsoft.com/cosmosdb/introducing-transactionalbatch-in-the-net-sdk/
MongoDB: https://www.mongodb.com/basics/transactions
DynamoDB: https://aws.amazon.com/blogs/aws/new-amazon-dynamodb-transactions/



I'm okay with removing the context on partitioning even though it's a core tenant of NoSQL (its the primary purpose behind the Dynamo, BigTable, Cassandra, HBase, et al projects) - but the original text should be removed because it is factually incorrect. Also think its a big miss to write an article on NoSQL and miss the primary reason for why many NoSQL systems were created in the first place (including why implemented Cosmos DB within Microsoft).

Reply:
Appreciate @aliuy's depth in NoSQL technologies. Think the best thing is for @aliuy and I to get on a call and craft the excat change. I'll reach out to him.

Reply:
Talked with @aliuy.  Let's go with the following:   

Moden NoSQL databases typically implement partitioning capabilities as a feature of their system design. Partition management is often built-in to the database, and routing is achieved through placement hints - often called partition keys. A flexible data models enables the NoSQL databases to lower the burden of schema management and improve availablity when deploying application updates that require data model changes.****